File: java/src/phase 1-conditional_thinking_java/Q21DistinctDigits.java
import java.util.Scanner;

/**
 * Phase 1: Conditional Thinking (If-Else, Boolean Logic)
 * Level 3: Math and Number Logic
 * Question 21: Take a 3-digit number and check if all digits are distinct.
 * 
 * Mentor Tips (Java 26):
 * - Extract digits using / and % operators. Let d1 = num / 100, d2 = (num / 10) % 10, d3 = num % 10. Check if d1 != d2 and d2 != d3 and d1 != d3.
 */
public class Q21DistinctDigits {

    public static boolean solve(int num) {
        // TODO: Implement this method to solve the question
        false;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("3-digit number: "); int num = scanner.nextInt();\n        System.out.println("Distinct: " + solve(num));
        scanner.close();
    }
}

File: python/phase 1-condtional_thinking/q21_distinct_digits.py
"""
Phase 1: Conditional Thinking (If-Else, Boolean Logic)
Level 3: Math and Number Logic
Question 21: Take a 3-digit number and check if all digits are distinct.

Mentor Tips (Python 3.14+):
- Extract digits using / and % operators. Let d1 = num / 100, d2 = (num / 10) % 10, d3 = num % 10. Check if d1 != d2 and d2 != d3 and d1 != d3.
"""

def solve(num: int) -> bool:
    # TODO: Implement this method to solve the question
    d1 = num // 100
    d2 = (num // 10) % 10
    d3 = num % 10
    return d1 != d2 and d2 != d3 and d1 != d3

def main():
    num = int(input("Enter a 3-digit number: "))
    print(f"Distinct: {solve(num)}")

if __name__ == "__main__":
    main()
